src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -54,7 +56,13 @@
     private final JsonFactory jsonFactory;
     private final ObjectReader eventReader;
 
+    private final Optional<MetricsCollector> metricsCollector;
```

Comment:
Yes, I was thinking of that solution as well but eventually decided for that `Optional` approach because it can deal with `null` values. In the end I don't care, I can change it.